homeassistant/components/zha/core/cluster_handlers/__init__.py

```diff
@@ -275,9 +275,33 @@ async def configure_reporting(self) -> None:
             try:
                 res = await self.cluster.configure_reporting_multiple(reports, **kwargs)
                 self._configure_reporting_status(reports, res[0])
-                # if we get a response, then it's a success
-                for attr_stat in event_data.values():
-                    attr_stat["success"] = True
+                # 2.5.8.1.3 Status Field
+                # The status field specifies the status of the Configure Reporting operation attempted on this attribute, as detailed in 2.5.7.3.
+                # Note that attribute status records are not included for successfully configured attributes, in order to save bandwidth.
+                # In the case of successful configuration of all attributes, only a single attribute status record SHALL be included in the command,
```

Comment:
To make sure I'm understanding ZCL correctly here, as I haven't tested this yet:

What happens if there are two successfully configured attributes and one failed?
The failed one is obviously included, but do the successfully ones show up both? Just one or not at all in that case?

Reply:
From that example, assuming just one or none of the successfully configured attributes are included in the report, the new code would display those as "failed" too, right?

Reply:
between what I included from the ZCL doc above and what is in referenced section 2.5.7.3:

```
When all attribute reporting configuration records have been processed, the device SHALL generate the constructed Configure Reporting Response command. If there are no attribute status records in the con- structed command, indicating that all attributes were configured successfully, a single attribute status record SHALL be included in the command, with the status field set to SUCCESS and the direction and attribute identifier fields omitted.
```

I think I assumed that in a mixed case scenario that we would get mixed individual report records in the response. I will check w/ a couple devices to see if this holds true.

Reply:
Good catch!!